paddle/fluid/operators/clip_by_norm_op.h

```diff
@@ -28,6 +31,22 @@ using SelectedRows = framework::SelectedRows;
 template <typename T, int MajorType = Eigen::RowMajor,
           typename IndexType = Eigen::DenseIndex>
 using EigenVector = framework::EigenVector<T, MajorType, IndexType>;
+template <typename T, int MajorType = Eigen::RowMajor,
+          typename IndexType = Eigen::DenseIndex>
+using EigenScalar = framework::EigenScalar<T, MajorType, IndexType>;
+
+template <typename T>
+struct ClipByNormFunctor {
+  ClipByNormFunctor(const T* scale, const T* x, T* out, const size_t& numel)
+      : scale_(scale), x_(x), out_(out), numel_(numel) {}
+  HOSTDEVICE inline void operator()(int64_t idx) const {
+    out_[numel_ - 1 - idx] = scale_[0] * x_[numel_ - 1 - idx];
+  }
```

Comment:
It seems that the implementation is problematic, you cannot make sure that out_[0] is updated last for GPU.